emu/src/internal_memory.rs

```diff
@@ -154,6 +159,34 @@ impl InternalMemory {
             _ => panic!("Reading an write-only memory address: {address:b}"),
         }
     }
+
+    fn write_address_timer_register(&mut self, address: u32, value: u8) {
+        match address {
+            0x04000100 => self.timer_registers.tm0cnt_l.set_byte(0, value),
+            0x04000102 => self.timer_registers.tm0cnt_h.set_byte(0, value),
+            0x04000104 => self.timer_registers.tm1cnt_l.set_byte(0, value),
+            0x04000106 => self.timer_registers.tm1cnt_h.set_byte(0, value),
+            0x04000108 => self.timer_registers.tm2cnt_l.set_byte(0, value),
+            0x0400010A => self.timer_registers.tm2cnt_h.set_byte(0, value),
+            0x0400010C => self.timer_registers.tm3cnt_l.set_byte(0, value),
+            0x0400010E => self.timer_registers.tm3cnt_h.set_byte(0, value),
+            _ => panic!("Reading an write-only memory address: {address:b}"),
+        }
+    }
+
+    fn read_address_timer_register(&self, address: u32) -> u8 {
+        match address {
+            0x04000100 => self.timer_registers.tm0cnt_l.read().get_byte(0),
+            0x04000102 => self.timer_registers.tm0cnt_h.read().get_byte(0),
+            0x04000104 => self.timer_registers.tm1cnt_l.read().get_byte(0),
+            0x04000106 => self.timer_registers.tm1cnt_h.read().get_byte(0),
+            0x04000108 => self.timer_registers.tm2cnt_l.read().get_byte(0),
+            0x0400010A => self.timer_registers.tm2cnt_h.read().get_byte(0),
+            0x0400010C => self.timer_registers.tm3cnt_l.read().get_byte(0),
+            0x0400010E => self.timer_registers.tm3cnt_h.read().get_byte(0),
```

Comment:
Why do we never read the higher part of the registers?

I would expect something like:

```rust
0x04000100 => self.timer_registers.tm0cnt_l.set_byte(0, value),
0x04000101 => self.timer_registers.tm0cnt_l.set_byte(1, value),
0x04000102 => self.timer_registers.tm0cnt_l.set_byte(0, value),
...
```

Maybe there's a reason why we never access these addresses?

Reply:
Ok I thing I got it why, it says that bits from 8 to 15 are not used, correct?

Reply:
![image](https://user-images.githubusercontent.com/99926684/198253473-9b9be6f1-4c00-4983-b736-f4b854ffde48.png)

Correct me if I'm wrong, but the bit from `8..-15` is not used.

Reply:
Yep, that's what I meant in the previous comment. I can't "resolve" this (and the 2 previous ones) comment, I think only you can?

Reply:
Yes, I think It works like this! :D

Reply:
What would you like to do? Resolve what?

Reply:
`Timer registers memory goes up to 0x0400010F since at 0x0400010E there's TM3CNT_H which is 2 bytes long` this and `Same as above`?

Reply:
Yes, sorry for the confusion :)

Reply:
No problem mate! :+1:

Reply:
I would love to see some comment in code for future doubt :)

Reply:
It's not a problem!